Xamarin.Forms.Platform.Android/Renderers/FontImageSourceHandler.cs

```diff
@@ -1,15 +1,17 @@
+using System.Collections.Generic;
 using System.IO;
 using System.Threading;
 using System.Threading.Tasks;
 using Android.Content;
 using Android.Graphics;
 using Android.Util;
-using Xamarin.Forms.Internals;
 
 namespace Xamarin.Forms.Platform.Android
 {
 	public sealed class FontImageSourceHandler : IImageSourceHandler
 	{
+		static Dictionary<string, Typeface> TypefaceCache = new Dictionary<string, Typeface>();
```

Comment:
Should this be a ConcurrentDictionary?  I'm pretty sure this code might be running on a task pool thread and dictionary isn't thread safe

Reply:
I was sure that this code would only be running on UI thread

Reply:
I used your solution and application was not stable. ConcurrentDictionary solve that.

Reply:
> I was sure that this code would only be running on UI thread

I don't think we can guarantee that (especially for the future).  The code that calls this has awaits on Tasks that have ConfigureAwait set to false

The main thing that made me lean towards using a ConcurrentDictionary for this is that this code is able to run on a Task Pool thread and still be functional.  I forced this block of code to run on a TaskPool thread and it still executed correctly so that tells me there's a future scenario where this might be the case and not having a thread safe dictionary feels like one of those hidden things that would bite us down the road.  If this code was only functional on the UI thread then I would say Dictionary would be fine.

Reply:
@PureWeen Ok, I've changed it to ConcurrentDictionary then, please check. thanks!